backends/openvino/runtime/OpenvinoBackend.cpp

```diff
@@ -0,0 +1,205 @@
+
+using namespace std;
+using executorch::aten::ScalarType;
+using executorch::runtime::ArrayRef;
+using executorch::runtime::Backend;
+using executorch::runtime::BackendExecutionContext;
+using executorch::runtime::BackendInitContext;
+using executorch::runtime::CompileSpec;
+using executorch::runtime::DelegateHandle;
+using executorch::runtime::Error;
+using executorch::runtime::EValue;
+using executorch::runtime::FreeableBuffer;
+using executorch::runtime::MemoryAllocator;
+using executorch::runtime::Result;
+
+namespace executorch {
+namespace backends {
+namespace openvino {
+
+OpenvinoBackend::OpenvinoBackend() {
+  if (!is_available()) {
+    // ET_LOG(Error, "OpenVINO runtime is not available. Initialization
+    // failed.");
+    throw std::runtime_error("OpenVINO runtime not available");
+  }
+
+  // ET_LOG(Info, "OpenVINO runtime successfully verified and initialized.");
+}
+
+bool OpenvinoBackend::is_available() const {
+  try {
+    // Create an OpenVINO Core object to verify runtime availability
+    ov::Core core;
+
+    // Check if at least one device is available
+    auto devices = core.get_available_devices();
+    if (!devices.empty()) {
+      return true; // OpenVINO is available
+    }
+  } catch (const std::exception& e) {
+    // Log the exception if OpenVINO runtime is not available
+    ET_LOG(Error, "OpenVINO is not available: %s", e.what());
+  } catch (...) {
+    // Handle any unexpected errors
+    ET_LOG(
+        Error, "OpenVINO availability check failed due to an unknown error.");
+  }
+
+  return false; // OpenVINO is not available
+}
+
+Result<DelegateHandle*> OpenvinoBackend::init(
+    BackendInitContext& context,
+    FreeableBuffer* processed,
+    ArrayRef<CompileSpec> compile_specs) const {
+  ET_LOG(Info, "OpenvinoBackend::init %p", processed->data());
+
+  ov::Core core;
+  const char* data_ptr = static_cast<const char*>(processed->data());
+  size_t data_size = processed->size();
+
+  // Copy data to a string or vector
+  std::string data_string(data_ptr, data_size);
+
+  // Wrap the data in a stream
+  std::istringstream compiled_stream(data_string);
+
+  auto device = "CPU";
+  // Get the device value, if provided in compile sepcs
+  for (auto& compile_spec : compile_specs) {
+    if (std::strcmp(compile_spec.key, "device") == 0)
+      device = static_cast<char*>(compile_spec.value.buffer);
+  }
+
+  // Import the model
+  auto compiled_model = core.import_model(compiled_stream, device);
+
+  // Allocate an infer request
+  std::shared_ptr<ov::InferRequest> infer_request =
+      std::make_shared<ov::InferRequest>(compiled_model.create_infer_request());
+
+  // Allocate execution handle
+  MemoryAllocator* allocator = context.get_runtime_allocator();
+  ExecutionHandle* handle =
+      ET_ALLOCATE_INSTANCE_OR_RETURN_ERROR(allocator, ExecutionHandle);
+  handle->compiled_model = std::make_shared<ov::CompiledModel>(compiled_model);
+  handle->infer_request = infer_request;
```

Comment:
Do you not want to free `processed` once consumed?

Reply:
We do not need `precessed` after consumed. We updated accordingly: https://github.com/ynimmaga/executorch/blob/0efa875b88def668a71d9357f4d75079e06f1e3c/backends/openvino/runtime/OpenvinoBackend.cpp#L77